### PR TITLE
Add World::ValidateGraphs method (sdf9)

### PR DESCRIPTION
**Pull request by scpeters**
# 🎉 New feature

I noticed a bug in ign-physics that needs a new API in libsdformat to properly fix it. This is similar to #601 but is targeted at `sdf9`.

## Summary

I noticed a test in ign-physics in which some models do not have properly loaded graphs, but it's hard to detect since the graph data is private. This creates a `World::ValidateGraphs()` method to call the hidden `validate*Graph` functions in FrameSemantics.cc  with the private graph data. It also requires an extra check in the `validate*Graph` functions that the scope points to a valid graph.

## Test it

Run the `UNIT_World_TEST` and `INTEGRATION_world_dom` tests and see that they pass.

## Checklist
- [X] Signed all commits for DCO
- [X] Added tests
- [ ] Added example and/or tutorial
- [X] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Code check passed (In source directory, run `sh tools/code_check.sh`)
- [ ] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review
[another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+)
to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge**

### CODE REVIEW

File: src/World_TEST.cc [hunk 1]
```diff
@@ -53,6 +53,17 @@ TEST(DOMWorld, Construction)
...
+  auto errors = world.ValidateGraphs();
+  EXPECT_EQ(2u, errors.size());
+  EXPECT_EQ(errors[0].Code(), sdf::ErrorCode::FRAME_ATTACHED_TO_GRAPH_ERROR);
+  EXPECT_NE(std::string::npos,
+    errors[0].Message().find(
+      "FrameAttachedToGraph pointer is null"));
+  EXPECT_EQ(errors[1].Code(), sdf::ErrorCode::POSE_RELATIVE_TO_GRAPH_ERROR);
+  EXPECT_NE(std::string::npos,
+    errors[1].Message().find(
+      "PoseRelativeToGraph pointer is null"));
```
Comment (chapulina) [inline src/World_TEST.cc:66]: Can we add some tests that have errors on the graphs themselves, not just missing graphs?

**Review by chapulina — COMMENTED** [review]
LGTM, just have a request for more testing.

I also noticed that codecov may not be correctly setup for this repo. Let's fix this next week.

File: src/World_TEST.cc (hunk 1 above)
Comment (scpeters) [reply src/World_TEST.cc:66]: a5985ab

**Review by chapulina — APPROVED** [review]

**Comment by scpeters** [comment]
there were some copy/paste issues with 82e2909a8d169feb681a5d1dbc8d82bac5d9cb7d so I replaced it with a5985abb104143db1d10e66bdfc186a1b389fb24

**Comment by scpeters** [comment]
I added `Model::ValidateGraphs` as well per https://github.com/ignitionrobotics/sdformat/pull/601#discussion_r661004493

**Review by azeey — APPROVED** [review]

**Pull request merged**